libpod/networking_linux.go

```diff
@@ -101,7 +101,15 @@ func (r *Runtime) configureNetNS(ctr *Container, ctrNS ns.NetNS) ([]*cnitypes.Re
 		requestedMAC = ctr.config.StaticMAC
 	}
 
-	podNetwork := r.getPodNetwork(ctr.ID(), ctr.Name(), ctrNS.Path(), ctr.config.Networks, ctr.config.PortMappings, requestedIP, requestedMAC)
+	var podName string
+	pod, err := r.GetPod(ctr.PodID())
```

Comment:
We should only make this call is `ctr.PodID() != ""` - this is a fairly expensive database lookup that we should avoid if we can.

Reply:
Looking at the boltdb code: https://github.com/containers/libpod/blob/598bb53d46dfc85b8bcc1e3000736106f80de93e/libpod/boltdb_state.go#L1366-L1369 it just return right away if the id is empty, so that shouldn't make a big difference.

Reply:
I don't want to force this decision on states, though - we aren't strictly wedded to Bolt, so a future SQLite-based state, for example, might not do that. Just a bit safer to do it here, IMO.